src/concrete_solve.jl

```diff
@@ -3,45 +3,58 @@
 # Here is where we can add a default algorithm for computing sensitivities
 # Based on problem information!
 
-function inplace_vjp(prob,u0,p)
+function inplace_vjp(prob,u0,p,has_cb,verbose)
   du = copy(u0)
   ez = try

```

Comment:
if we know it has issues with callbacks, then we should do `!has_cb` checks before we try it.

Reply:
can you isolate the enzyme issue?

Reply:
It happens in: 
https://github.com/SciML/DiffEqSensitivity.jl/blob/e4fcd60b8bf9853b3d79134a4643043a410ba4ea/src/derivative_wrappers.jl#L471
(Should this be `S.f` here?)

coming from 
https://github.com/SciML/DiffEqSensitivity.jl/blob/e4fcd60b8bf9853b3d79134a4643043a410ba4ea/src/callback_tracking.jl#L278

with the new `CallbackSensitivityFunction`
https://github.com/SciML/DiffEqSensitivity.jl/blob/master/src/callback_tracking.jl#L200-L203

Reply:
`S.f` fails a few lines later https://github.com/SciML/DiffEqSensitivity.jl/blob/e4fcd60b8bf9853b3d79134a4643043a410ba4ea/src/derivative_wrappers.jl#L501
because of `Enzyme.Duplicated(y, tmp1)`
```julia
julia> typeof(y)
SubArray{Float64, 1, Vector{Float64}, Tuple{UnitRange{Int64}}, true}

julia> typeof(tmp1)
Vector{Float64} (alias for Array{Float64, 1})
```
So 
```
struct CallbackSensitivityFunction{fType,Alg<:DiffEqBase.AbstractSensitivityAlgorithm,C<:AdjointDiffCache,pType} <: SensitivityFunction
    f::fType
    sensealg::Alg
    diffcache::C
    prob::pType
end
```
should probably get a `y` of `u` type. 

```julia
ERROR: LoadError: MethodError: no method matching Enzyme.Duplicated(::SubArray{Float64, 1, Vector{Float64}, Tuple{UnitRange{Int64}}, true}, ::Vector{Float64})
Closest candidates are:
  Enzyme.Duplicated(::T, ::T) where T at ~/.julia/packages/Enzyme/7MHm8/src/Enzyme.jl:66
Stacktrace:
  [1] _vecjacobian!(dλ::SubArray{Float64, 1, Vector{Float64}, Tuple{UnitRange{I
```

Reply:
Open an issue on this? It's not clear to me what causes this, but this is probably an important one because Enzyme should be a good default here.